src/lib/net/src/packets/incoming/keep_alive.rs

```diff
@@ -1,27 +1,33 @@
-use crate::packets::outgoing::keep_alive::KeepAlive;
+use crate::packets::outgoing::keep_alive::OutgoingKeepAlive;
 use crate::packets::IncomingPacket;
+use crate::utils::state::terminate_connection;
 use crate::{NetResult, ServerState};
 use ferrumc_macros::{packet, NetDecode};
 use std::sync::Arc;
 use tracing::debug;
 
 #[derive(NetDecode)]
 #[packet(packet_id = 0x18, state = "play")]
-pub struct IncomingKeepAlivePacket {
+pub struct IncomingKeepAlive {
```

Comment:
Please keep packets have a prefix `Packet` since they're publicly exported and it'd be confusing.